core/SettingsPiwik.php

```diff
@@ -481,7 +481,7 @@ public static function getPiwikInstanceId()
 
         $instanceId = GeneralConfig::getConfigValue('instance_id');
         if (!empty($instanceId)) {
-            return $instanceId;
+            return preg_replace('/[^\w\.-]/', '', $instanceId);
```

Comment:
@tsteur is this set of allowed characters, or shall we allow any other special character?

Reply:
Looking at https://domainname.shop/faq?id=7 I believe that should be fine. Technically `_` is not allowed but that's no problem.